File: app/src/main/java/vn/edu/usth/minigh/BaseActivity.java
// Base activity with drawer
package vn.edu.usth.minigh;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;


public class BaseActivity extends AppCompatActivity {
    DrawerLayout drawerLayout;

    public BaseActivity(int contentLayoutId) {
        super(contentLayoutId);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        drawerLayout = findViewById(R.id.drawer_layout);

        TextView follower = (TextView) findViewById(R.id.follower_total);
        follower.setText(Html.fromHtml("<b>"+5+"</b> Follower"));

        TextView following = (TextView) findViewById(R.id.following_total);
        following.setText((Html.fromHtml("<b>"+5+"</b> Following")));
    }

    @Override
    protected void onPause() {
        super.onPause();
        closeDrawer(drawerLayout);
    }

    public void openDrawer(View view) {
        drawerLayout.openDrawer(GravityCompat.START);
    }

    public void closeDrawer(View view) {
        if(drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START);
        }
    }

    public void redirectActivity(Class aclass) {
        Intent intent = new Intent(this, aclass);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }

    public void clickProfile(View view) {
        if (this.getClass() != ProfileActivity.class) {
            redirectActivity(ProfileActivity.class);
        }
    }

    public void clickRepo(View view) {
        if (this.getClass() != RepoActivity.class) {
            redirectActivity(RepoActivity.class);
        }
    }
    public void clickIssue(View view){
        if (this.getClass() != IssueActivity.class) {
            redirectActivity(IssueActivity.class);
        }
    }
    public void clickPRs(View view){
        if (this.getClass() != PRActivity.class) {
            redirectActivity(PRActivity.class);
        }
    }

    public void logout(View view) {
        AlertDialog.Builder builder = new AlertDialog.Builder(this);
        builder.setTitle("Log out");
        builder.setMessage("Are you sure you want to log out?");
        builder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                SessionManager manager = new SessionManager(
                    getApplicationContext());
                manager.removeSession();
                redirectActivity(AuthActivity.class);
            }
        });
        builder.setNegativeButton("No", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dialog.dismiss();
            }
        });
        builder.show();
    }
}

File: app/src/main/java/vn/edu/usth/minigh/BaseActivity.kt
// Base activity with drawer
package vn.edu.usth.minigh

import android.app.AlertDialog
import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.text.Html
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.lifecycle.lifecycleScope
import coil.load
import kotlinx.coroutines.launch
import vn.edu.usth.minigh.api.github
import java.net.CookieManager

abstract class BaseActivity(layoutID: Int) : AppCompatActivity(layoutID) {
    private lateinit var drawerLayout: DrawerLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        drawerLayout = findViewById(R.id.drawer_layout)
        val token = TokenManager(applicationContext).getToken()

        val avatar = findViewById<ImageView>(R.id.avatar)
        val username = findViewById<TextView>(R.id.nav_username)
        val follower = findViewById<TextView>(R.id.follower_total)
        val following = findViewById<TextView>(R.id.following_total)

        lifecycleScope.launch {
            val user = github.current_user("Bearer $token") // TODO: Replace this with login
            username.text = user.login
            follower.text = Html.fromHtml("<b>${user.followers}</b> Follower")
            following.text = (Html.fromHtml("<b>${user.following}</b> Following"))
            avatar.load(user.avatar_url)
        }
    }

    override fun onPause() {
        super.onPause()
        if (drawerLayout.isDrawerOpen(GravityCompat.START)) {
            drawerLayout.closeDrawer(GravityCompat.START)
        }
    }

    fun redirectActivity(aclass: Class<*>) {
        if (this::class.java == aclass) {
            drawerLayout.closeDrawer(GravityCompat.START)
        } else {
            startActivity(Intent(this, aclass))
        }
    }

    fun clickProfile(view: View) = redirectActivity(ProfileActivity::class.java)
    fun clickIssues(view: View) = redirectActivity(IssueActivity::class.java)
    fun clickPRs(view: View) = redirectActivity(PRActivity::class.java)
    fun openDrawer(view: View) = drawerLayout.openDrawer(GravityCompat.START)

    fun logout(view: View) {
        // TODO: define layout statically
        val builder = AlertDialog.Builder(this)
        builder.setTitle("Log out")
        builder.setMessage("Are you sure you want to log out?")
        builder.setPositiveButton(
            "Yes", DialogInterface.OnClickListener { _, _ ->
                SessionManager(applicationContext).removeSession()
                TokenManager(applicationContext).removeToken()
                redirectActivity(AuthActivity::class.java)
            }
        )
        builder.setNegativeButton(
            "No", DialogInterface.OnClickListener { dialog, _ ->
                dialog.dismiss()
            }
        )
        builder.show()
    }
}
